src/components/MaLink/MaLink.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+  <a :class="[linkStyles, 'ma-link']" :href="href"><slot /></a>
+</template>
+
+<script>
+export default {
+  name: 'MaLink',
+
+  props: {
+    /**
+     * Sets the component displaying style
+     */
+    category: {
+      type: String,
+      default: 'primary',
+      validator: (v) =>
+        ['primary', 'secondary', 'white', 'gradient', 'no-background'].includes(
+          v
+        ),
+    },
+
+    href: {
+      type: String,
+      required: true,
+    },
+  },
+
+  computed: {
+    linkStyles() {
+      return {
+        [`ma-link--${this.category}`]: this.category,
+      }
+    },
+  },
```

Comment:
why not interpolating directly in the template?

```html
<a :class="`ma-link--${ category } ma-link`" :href="href"><slot /></a>
```

Or at least making this computed a bit more clear

```suggestion
  computed: {
    cssClass() {
      return `ma-link--${this.category}`
    },
  },
```

Reply:
ok, i'll go ahead and make the change, but since this is a stylistic change I'd rather have a linter rule to check against or leave it to the discretion of authors xD